obspy/dyna/core.py

```diff
@@ -0,0 +1,827 @@
+    fh.write("FILTER_ORDER: %s\n" %
+             floattostr(stream[0].stats.dyna.FILTER_ORDER, 0))
+    fh.write("LOW_CUT_FREQUENCY_HZ: %s\n" %
+             floattostr(stream[0].stats.dyna.LOW_CUT_FREQUENCY_HZ, 3))
+    fh.write("HIGH_CUT_FREQUENCY_HZ: %s\n" %
+             floattostr(stream[0].stats.dyna.HIGH_CUT_FREQUENCY_HZ, 3))
+    fh.write("LATE/NORMAL_TRIGGERED: %s\n" %
+             stream[0].stats.dyna.LATE_NORMAL_TRIGGERED)
+    fh.write("DATABASE_VERSION: %s\n" % stream[0].stats.dyna.DATABASE_VERSION)
+    fh.write("HEADER_FORMAT: DYNA 1.0\n")
+    fh.write("DATA_TYPE: %s\n" % stream[0].stats.dyna.DATA_TYPE)
+    fh.write("PROCESSING: %s\n" % stream[0].stats.dyna.PROCESSING)
+    fh.write("DATA_TIMESTAMP_YYYYMMDD_HHMMSS: %s\n" %
+             stream[0].stats.dyna.DATA_TIMESTAMP_YYYYMMDD_HHMMSS)
+    fh.write("USER1: %s\n" % stream[0].stats.dyna.USER1)
+    fh.write("USER2: %s\n" % stream[0].stats.dyna.USER2)
+    fh.write("USER3: %s\n" % stream[0].stats.dyna.USER3)
+    fh.write("USER4: %s\n" % stream[0].stats.dyna.USER4)
+
+    if stream[0].stats.dyna.DATA_TYPE in ("ACCELERATION", "VELOCITY"):
+        for d in stream[0].data:
+            fh.write("%-.6f\n" % d)
+    elif stream[0].stats.dyna.DATA_TYPE == "DISPLACEMENT":
+        for d in stream[0].data:
+            fh.write("%e\n" % d)
+    elif stream[0].stats.dyna.DATA_TYPE[-8:] == "SPECTRUM":
+        for j in range(len(stream[0].data)):
+            fh.write("%12.6f" % stream[0].data[j])
+            fh.write("%13.6f\n" % stream[1].data[j])
+
+    fh.close()
+
+
+def toUTCDateTime(value):
+    """
+    Converts time string used within DYNA/ITACA File into a UTCDateTime.
+
+    :type value: str
+    :param value: A Date time string.
+    :return: Converted :class:`~obspy.core.UTCDateTime` object.
+
+    .. rubric:: Example
+
+    >>> toUTCDateTime('20090330_215650.967')
+    UTCDateTime(2009, 3, 30, 21, 56, 50, 967000)
+    """
+    try:
+        date, time = value.split('_')
+    except ValueError:
+        date = value
+
+    year = int(date[0:4])
+    month = int(date[4:6])
+    day = int(date[6:8])
+
+    hour = int(time[0:2])
+    mins = int(time[2:4])
+    secs = float(time[4:])
+
+    return UTCDateTime(year, month, day, hour, mins) + secs
+
+
+def fromUTCDateTime(dt):
+    """
+    Converts UTCDateTime object into a time string used within Seismic Handler.
+
+    :type dt: :class:`~obspy.core.UTCDateTime`
+    :param dt: A UTCDateTime object.
+    :return: Converted date time string as defined in DYNA file header.
+
+    .. rubric:: Example
+
+    >>> from obspy.core import UTCDateTime
+    >>> dt = UTCDateTime(2009, 3, 30, 21, 56, 50, 967000)
+    >>> fromUTCDateTime(dt)
+    '20090330_215650.967'
+    """
+    pattern = "%4d%02d%02d_%02d%02d%02d.%03d"
+
+    return pattern % (dt.year, dt.month, dt.day, dt.hour,
+                      dt.minute, dt.second, dt.microsecond / 1000)
+
+
+def strtofloat(sf):
```

Comment:
Yes, probably only empty values should pass. Maybe non-existing header fields with non-existing values should not be inserted at all.